41-nautilus/technical_specification.md

```diff
@@ -0,0 +1,83 @@
+# Refactoring Priorities (Nautilus)
+**Epic Type:** Implementation/Exploratory Epic
+
+Epic planning and implementation follows the
+[Epic Planning and Marathon SOP](https://docs.ghga-dev.de/main/sops/sop001_epic_planning.html).
+
+**Attention: Please do not put any confidential content here.**
+
+## Scope
+### Outline:
+This epic aims to explore and implement some of the refactoring priorities that are simple to implement
+but give especially high benefits.
+
+
+### Included/Required:
+- replace dependency injection framework
+- make devcontainer environment more lightweight
+- streamline CI and local workflows
+
+
+## Additional Details:
+
+### Replace Dependency Injection Framework
+
+Currently, we are using the dependency_injector library and are experiencing the following problems:
+- difficult debugability:
+    - e.g. if arguments are not passed correctly to the constructors, often the container just freezes without
+    indications on where the problem is
+    - the debugger cannot enter the cython code
+- initialization does not happen lazy, i.e. event consumers are also started when starting the REST API of a service
```

Comment:
I don't think that's a general flaw of the dependency_injector library, but a mere consequence of how our own AsyncContructor works based on the Resource provider class. This could probably be improved, maybe basing it on the Singleton or Factory provider class instead of the Resource provider class.

Reply:
another problem:
- single developer, many open issues, maintenance unclear